CDS/js/add-asset-lib.js

```diff
@@ -0,0 +1,62 @@
+var reqwest = require('reqwest');
+var Promise = require('promise');
+var datastore = require('./Datastore');
+var hmac = require('./hmac');
+
+const urlBase = process.env.url_base;
+const path = '/api2/atom/:id/asset';
+
+function postAsset() {
+
+    if (!process.env.url_base) {
+        return new Promise((fulfill, reject) => {
+            reject(new Error('Cannot add assets to media atom: missing url base'))
+        });
+    }
+    if (!process.env.atom_id) {
+        return new Promise((fulfill, reject) => {
+            reject(new Error('Cannot add assets to media atom: missing atom id'))
+        });
+    }
+
+    var connection = new datastore.Connection('AddAAsset');
+    datastore.initialiseDb();
+    let urlBase, atomId;
+
+    return datastore.substituteStrings(connection, [process.env.url_base, process.env.atom_id])
+    .then(substitutedStrings => {
+        [urlBase, atomId] = substitutedStrings;
+
+        return datastore.get(connection, 'meta', 'youtube_url')
+        .then(result => {
+            const date = (new Date()).toUTCString();
+
+            const youtubeUrl = result.value;
+
+            const data = { uri: youtubeUrl };
+            const uri = path.replace(/:id/, atomId);
+            const url = urlBase + uri;
+
+            return hmac.makeHMACToken(connection, date, uri)
+            .then(token => {
+
+                return reqwest({
+                    url: url,
+                    method: 'POST',
```

Comment:
about to release a PR to change this endpoint to a `PUT` rather than `POST` - you'll need to update your PR unfortunately.